new-packages/ts-project/src/state.ts

```diff
@@ -221,6 +222,16 @@ function registerActionBlocks(
   }
 }
 
+function getDefaultInternalValue(
+  ctx: ExtractionContext,
+  targets: string[] | undefined,
+) {
+  if (ctx.version !== 'v4') {
```

Comment:
Why isn't the version a number? `version: 4 | 5`.

Reply:
it just felt nicer to me to be able to potentially extend this in the future with stuff like `'v5.10'` instead of `5.1` (`v5.10` is not representable as a number since it's the same as `5.1`)

Reply:
Are you thinking a runtime/extraction difference in non-major XState versions?

Reply:
yes

Reply:
the runtime/extraction behavior etc should rather stay the same within the major but we might want to raise errors differently or something

Reply:
I'd drop the v because it's given by the property name but this is nit.